File: Python/7_Binary Search/BinarySearch.py
def binary_search(my_list, l, r, x):
	if r >= l:
		mid = l + (r - l)//2
		if my_list[mid] == x:
			return mid
		elif my_list[mid] > x:
			return binary_search(my_list, l, mid-1, x)
		else:
			return binary_search(my_list, mid+1, r, x)
	else:
		return -1

nums = [2, 3, 4, 10, 40]
x = 10

print(binary_search(nums, 0, len(nums)-1, x))

File: Ruby/7_Binary Search/Binary Search.rb
def binarySearch(arr, l, r, x)
	if l <= r
		mid = (l + r) / 2
		if arr[mid] > x 
			binarySearch(arr, l, mid - 1, x)
		elsif arr[mid] < x
			binarySearch(arr, mid + 1, r, x)
		else
			return mid
		end
	end
end

nums = [18, 27, 36, 45, 59]
puts binarySearch(nums, 0, nums.length - 1, 59)